src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -123,8 +128,28 @@ private String setPlayerTeamToTeleporterTeam(EntityRef player, EntityRef telepor
 
     private void handlePlayerTeleport(EntityRef player, String team) {
         Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
+        player.send(new PregameEvent());
         player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
         player.addOrSaveComponent(startingInventory);
         player.send(new RequestInventoryEvent(startingInventory.items));
+        sendGameStartMessageEventToClients();
+    }
+
+    private void sendTimerEventToClients() {
+        if (entityManager.getCountOfEntitiesWith(ClientComponent.class) != 0) {
```

Comment:
Why do you need this `if` check? If there are no clients, the iterable should be empty, and the `for` loop should effectively be a no-op.